### PR TITLE
Migrate docs to FullNodeClient

### LINKED ISSUE #1089: Migrate docs to `FullNodeClient`
### Feature Request

With deprecation of GatewayClient, docs should no longer use GatewayClient as the main, preferred solution (although GatewayClient should still stay as an option).

### PR DESCRIPTION
**Pull request by tomek0123456789**
Closes #1089 

## Introduced changes

- 
- 

##

- [ ] This PR contains breaking changes

### CODE REVIEW

**Review by drknzz — COMMENTED** [review]
👍 

One more place I found in docs that could use some rewording:
https://starknetpy--1092.org.readthedocs.build/en/1092/guide/account_and_client.html#fullnodeclient-usage

File: starknet_py/tests/e2e/docs/code_examples/test_contract.py [hunk 1]
```diff
@@ -24,12 +24,12 @@ def test_init():
         ],
         provider=Account(
             address=0x321,
-            client=GatewayClient(TESTNET),
+            client=FullNodeClient(TESTNET),
```
[the remark this answers is not in the document]
Comment (tomek0123456789) [reply starknet_py/tests/e2e/docs/code_examples/test_contract.py:27]: Good catch, thanks!

[the remark this answers is not in the document]
Comment (tomek0123456789) [reply starknet_py/tests/e2e/docs/code_examples/test_contract.py:27]: Good catch, thanks!

File: starknet_py/tests/e2e/docs/guide/test_cairo1_contract.py [hunk 2]
```diff
@@ -13,23 +12,17 @@
...
-    )
+    ) = sierra_minimal_compiled_contract_and_class_hash
 
     contract_compiled_casm = read_contract(
-        "minimal_contract_compiled.casm"
-        if isinstance(account.client, GatewayClient)
-        else "another_minimal_contract_compiled.casm",
+        "minimal_contract_compiled.casm",
         directory=CONTRACTS_COMPILED_V1_DIR,
     )
```
Comment (drknzz) [inline starknet_py/tests/e2e/docs/guide/test_cairo1_contract.py:27]: I was wondering for a second how come this works, but realised we're running full node and gateway separately 👍

**Review by drknzz — APPROVED** [review]

File: docs/guide/account_and_client.rst [hunk 3]
```diff
@@ -43,14 +43,14 @@ Note that the nonce will be bumped only by 1.
...
+Use a :ref:`FullNodeClient` to interact with services providing `Starknet RPC interface <https://github.com/starkware-libs/starknet-specs/blob/606c21e06be92ea1543fd0134b7f98df622c2fbf/api/starknet_api_openrpc.json>`_
+like `Pathfinder <https://github.com/eqlabs/pathfinder>`_,
+`Papyrus <https://github.com/starkware-libs/papyrus>`_, `Juno <https://github.com/NethermindEth/juno>`_
+or `starknet-devnet <https://github.com/0xSpaceShard/starknet-devnet>`_.
 Using own full node allows for querying Starknet with better performance.
-Since gateway will be deprecated at some point in the future, having ``FullNodeClient`` with interface uniform with that of ``GatewayClient``
-will allow for simple migration for starknet.py users.
+
+Since GatewayClient is deprecated and will be removed at some point in the future, having ``FullNodeClient``
+with interface uniform with that of ``GatewayClient`` will allow for simple migration for starknet_py users.
```
Comment (drknzz) [inline docs/guide/account_and_client.rst:53]:
```suggestion
with interface uniform with that of ``GatewayClient`` will allow for simple migration for starknet.py users.
```

**Pull request merged**